# BEST PERFORMING MOVIES AT THE BOX OFFICE ANALYSIS

### OVERVIEW*

### BUSINESS UNDERSTANDING*

Microsoft has ventured into the movie production industry. Being new , they lack expertise on film production among them audience preference and industry patterns. In this project I explore the elements that influence and determine the high consistent box office earnings attained by high performing films. This can be used to inform decisions on which films to create. 

### DATA UNDERSTANDING

The movies datasets are from Box Office Mojo, IMDb, Rotten Tomatoes, Rotten Tomatoes Reviews , The MovieDB and The Numbers

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import sqlite3

In [2]:
box_office_mojo_df = pd.read_csv("bom.movie_gross.csv")
rotten_tomatoes_df = pd.read_csv("rt.movie_info.tsv",  delimiter='\t')
rt_reviews_df = pd.read_csv("rt.reviews.tsv", delimiter='\t', encoding='ISO-8859-1') 
tmdb_df = pd.read_csv("tmdb.movies.csv", index_col=0)
tn_movie_budgets_df = pd.read_csv("tn.movie_budgets.csv") 

#### Box Office Mojo

The dataset comprises information on a total of 3,387 movies, featuring details such as movie titles, studio names, domestic gross earnings, foreign gross earnings, and release years for each individual film.

In [3]:
box_office_mojo_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [4]:
box_office_mojo_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [5]:
box_office_mojo_df.tail()

title      studio  domestic_gross foreign_gross  \
3382                    The Quake       Magn.          6200.0           NaN   
3383  Edward II (2018 re-release)          FM          4800.0           NaN   
3384                     El Pacto        Sony          2500.0           NaN   
3385                     The Swan  Synergetic          2400.0           NaN   
3386            An Actor Prepares       Grav.          1700.0           NaN   

      year  
3382  2018  
3383  2018  
3384  2018  
3385  2018  
3386  2018

In [6]:
box_office_mojo_df.shape

(3387, 5)

In [7]:
box_office_mojo_df.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [8]:
box_office_mojo_df["title"].value_counts()

Bluebeard                                                      2
Beloved Sisters                                                1
Detective Dee and the Mystery of the Phantom Flame             1
Rampart                                                        1
No (2013)                                                      1
                                                              ..
Shootout at Wadala                                             1
Begin Again                                                    1
Skyscraper                                                     1
Argento's Dracula 3D                                           1
Alexander and the Terrible, Horrible, No Good, Very Bad Day    1
Name: title, Length: 3386, dtype: int64

In [9]:
box_office_mojo_df.duplicated().sum()

0

In [10]:
box_office_mojo_df[box_office_mojo_df["title"] == 'Bluebeard']


title  studio  domestic_gross foreign_gross  year
317   Bluebeard  Strand         33500.0          5200  2010
3045  Bluebeard   WGUSA         43100.0           NaN  2017

There are no duplicates. Bluebeard movie appears twice and has been produced by different studios. 

In [11]:
box_office_mojo_df["studio"].value_counts()

IFC      166
Uni.     147
WB       140
Magn.    136
Fox      136
        ... 
Linn       1
IVP        1
TFC        1
CP         1
KC         1
Name: studio, Length: 257, dtype: int64

There are missing values on domestic gross, and foreign gross columns. 

In [12]:
box_office_mojo_df["domestic_gross"].value_counts()

1100000.0      32
1000000.0      30
1300000.0      30
1200000.0      25
1400000.0      23
               ..
68800.0         1
87000000.0      1
739000.0        1
336000000.0     1
727000.0        1
Name: domestic_gross, Length: 1797, dtype: int64

In [13]:
box_office_mojo_df["foreign_gross"].value_counts()

1200000      23
1100000      14
1900000      12
4200000      12
2500000      11
             ..
185500000     1
305000        1
24300000      1
149000        1
100000000     1
Name: foreign_gross, Length: 1204, dtype: int64

In [14]:
box_office_mojo_df["year"].value_counts()

2015    450
2016    436
2012    400
2011    399
2014    395
2013    350
2010    328
2017    321
2018    308
Name: year, dtype: int64

####  The Numbers Dataset

In [15]:
tn_movie_budgets_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


The dataset has no nulls

In [16]:
tn_movie_budgets_df.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [17]:
tn_movie_budgets_df["movie"].value_counts()

Halloween                                         3
King Kong                                         3
Home                                              3
Ghostbusters                                      2
Poltergeist                                       2
                                                 ..
Night at the Museum: Battle of the Smithsonian    1
Margaret                                          1
Nocturnal Animals                                 1
Por amor en el caserio                            1
She's All That                                    1
Name: movie, Length: 5698, dtype: int64

In [18]:
tn_movie_budgets_df["release_date"].value_counts()

Dec 31, 2014    24
Dec 31, 2015    23
Dec 31, 2010    15
Dec 31, 2008    14
Dec 31, 2009    13
                ..
Mar 8, 1996      1
Jul 1, 2010      1
Oct 16, 1996     1
Dec 31, 1996     1
Mar 30, 1988     1
Name: release_date, Length: 2418, dtype: int64

In [19]:
tn_movie_budgets_df["production_budget"].value_counts()

$20,000,000    231
$10,000,000    212
$30,000,000    177
$15,000,000    173
$25,000,000    171
              ... 
$1,644,000       1
$777,000         1
$30,250,000      1
$13,200,000      1
$49,800,000      1
Name: production_budget, Length: 509, dtype: int64

In [20]:
tn_movie_budgets_df["domestic_gross"].value_counts()

$0             548
$8,000,000       9
$2,000,000       7
$7,000,000       7
$10,000,000      6
              ... 
$19,019,882      1
$80,571,655      1
$17,583,468      1
$4,930           1
$39,175,066      1
Name: domestic_gross, Length: 5164, dtype: int64

In [21]:
tn_movie_budgets_df["worldwide_gross"].value_counts()

$0             367
$8,000,000       9
$2,000,000       6
$7,000,000       6
$10,000,000      4
              ... 
$261,364         1
$6,626,115       1
$76,657,000      1
$35,417,162      1
$57,907,734      1
Name: worldwide_gross, Length: 5356, dtype: int64

There are movies with the same title but are not duplicates. 

#### Rotten Tomatoes Info

In [22]:
rotten_tomatoes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


There dataset has 1560 movies with features such as the id, synopsis, rating, genre, director, writer,theater date, dvd_date, currency, box_office, runtime, studio
All columns bar the id have null values

#### Rotten Tomatoes Reviews

In [23]:
rt_reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


There dataset comprises of 54432 reviews. The reviews feature id, review, rating, fresh, critic, top_critic, publisher and date. 

#### The Movie Dataset

In [24]:
tmdb_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


There are no null values.

In [25]:
tmdb_df["genre_ids"].value_counts()

[99]                   3700
[]                     2479
[18]                   2268
[35]                   1660
[27]                   1145
                       ... 
[80, 18, 10749, 53]       1
[27, 9648, 14, 28]        1
[16, 28, 35, 9648]        1
[27, 28, 12, 878]         1
[35, 878, 28]             1
Name: genre_ids, Length: 2477, dtype: int64

In [26]:
tmdb_df["original_language"].value_counts()

en    23291
fr      507
es      455
ru      298
ja      265
      ...  
sw        1
cr        1
nb        1
ha        1
af        1
Name: original_language, Length: 76, dtype: int64

In [27]:
tmdb_df["title"].value_counts()

Eden                                                                   7
Home                                                                   7
Alone                                                                  5
The Box                                                                5
Truth or Dare                                                          5
                                                                      ..
Mud                                                                    1
They Won't Stay Dead: A Look at 'Return of the Living Dead Part II'    1
God's Country                                                          1
i49 TF2 Fragumentary                                                   1
Led Zeppelin: Dazed & Confused                                         1
Name: title, Length: 24688, dtype: int64

In [28]:
tmdb_df["release_date"].value_counts()

2010-01-01    269
2011-01-01    200
2012-01-01    155
2014-01-01    155
2013-01-01    145
             ... 
1967-03-06      1
2012-08-30      1
1946-12-25      1
2015-01-19      1
2005-11-22      1
Name: release_date, Length: 3433, dtype: int64

In [29]:
tmdb_df["popularity"].value_counts()

0.600     7037
1.400      649
0.840      587
0.624      104
0.625       92
          ... 
3.742        1
14.749       1
7.924        1
8.414        1
9.060        1
Name: popularity, Length: 7425, dtype: int64

#### IMDb

In [30]:
conn = sqlite3.connect("im.db")

In [69]:
# query = """
#     SELECT primary_title, start_year, runtime_minutes, genres
#     FROM movie_basics
#     WHERE runtime_minutes IS NOT NULL AND genres IS NOT NULL
# """
query2 = """
    SELECT mb.primary_title, mb.start_year, mb.runtime_minutes, mb.genres, mv.averagerating AS rating, numvotes AS votes
    FROM movie_basics mb
    JOIN movie_ratings mv
    ON mb.movie_id = mv.movie_id
    WHERE runtime_minutes IS NOT NULL AND genres IS NOT NULL
"""
# IMDb_df = pd.read_sql(query, conn).isna().any()
IMDb_popularity = pd.read_sql(query2, conn)
IMDb_popularity.head()

primary_title  start_year  runtime_minutes  \
0                        Sunghursh        2013            175.0   
1  One Day Before the Rainy Season        2019            114.0   
2       The Other Side of the Wind        2018            122.0   
3         The Wandering Soap Opera        2017             80.0   
4                  Joe Finds Grace        2017             83.0   

                       genres  rating  votes  
0          Action,Crime,Drama     7.0     77  
1             Biography,Drama     7.2     43  
2                       Drama     6.9   4517  
3        Comedy,Drama,Fantasy     6.5    119  
4  Adventure,Animation,Comedy     8.1    263

### Data Cleaning

#### Box Office Mojo
Check and deal with NaN and missing values

In [32]:
# Check for NaN values in domestic_gross column
box_office_mojo_df["domestic_gross"].isna().sum()

28

In [33]:
# drop the columns with NaN values in domestic_gross column
box_office_mojo_df.dropna(subset=["domestic_gross"], inplace=True)

In [34]:
# Check for NaN values in foreign gross column

box_office_mojo_df["foreign_gross"].isna().sum()

1350

In [35]:
# drop the columns with NaN values in foreign_gross column

box_office_mojo_df.dropna(subset=["foreign_gross"], inplace=True)

In [36]:
box_office_mojo_df["studio"].isna().sum()

2

In [37]:
# There are only two data without studios, we can drop them
box_office_mojo_df.dropna(subset=["studio"], inplace=True)

In [38]:
# confirm there are no missing values in the dataset
box_office_mojo_df.isna().any()

title             False
studio            False
domestic_gross    False
foreign_gross     False
year              False
dtype: bool

There are no missing values.

The foreign column should be changed to a float datatype

In [39]:
box_office_mojo_df.dtypes

title              object
studio             object
domestic_gross    float64
foreign_gross      object
year                int64
dtype: object

In [40]:
# replace the commas and the change to a float type
box_office_mojo_df["foreign_gross"] = box_office_mojo_df["foreign_gross"].str.replace(',', '').astype(float)
box_office_mojo_df.dtypes

title              object
studio             object
domestic_gross    float64
foreign_gross     float64
year                int64
dtype: object

In [41]:
box_office_mojo_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2007 entries, 0 to 3353
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           2007 non-null   object 
 1   studio          2007 non-null   object 
 2   domestic_gross  2007 non-null   float64
 3   foreign_gross   2007 non-null   float64
 4   year            2007 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 94.1+ KB


#### The Numbers

This dataset has no null values but has placeholders in the domestic gross and worldwide gross.

In [42]:
# The missing values are denoted by $0
tn_movie_budgets_df["domestic_gross"].value_counts()

$0             548
$8,000,000       9
$2,000,000       7
$7,000,000       7
$10,000,000      6
              ... 
$19,019,882      1
$80,571,655      1
$17,583,468      1
$4,930           1
$39,175,066      1
Name: domestic_gross, Length: 5164, dtype: int64

In [43]:
tn_movie_budgets_df["worldwide_gross"].value_counts()

$0             367
$8,000,000       9
$2,000,000       6
$7,000,000       6
$10,000,000      4
              ... 
$261,364         1
$6,626,115       1
$76,657,000      1
$35,417,162      1
$57,907,734      1
Name: worldwide_gross, Length: 5356, dtype: int64

In [44]:
# Drop the  rows with $0 in domestic gross and worldwide gross 
tn_movie_budgets_df = tn_movie_budgets_df[(tn_movie_budgets_df["domestic_gross"] != '$0') | (tn_movie_budgets_df["worldwide_gross"] != '$0')]

In [45]:
tn_movie_budgets_df["worldwide_gross"].value_counts()

$8,000,000     9
$2,000,000     6
$7,000,000     6
$5,000,000     4
$4,000,000     4
              ..
$6,626,115     1
$76,657,000    1
$35,417,162    1
$51,957        1
$57,907,734    1
Name: worldwide_gross, Length: 5355, dtype: int64

Change the domestic and worlwide gross columns to float

In [46]:
def floating(amount):
    new_amount = None
    if type(amount) == str:
        new_amount = float(amount.strip('$').replace(',',''))
    else:
        new_amount = float(amount)

    return new_amount
tn_movie_budgets_df["domestic_gross"] = tn_movie_budgets_df["domestic_gross"].map(floating)
tn_movie_budgets_df["worldwide_gross"] = tn_movie_budgets_df["worldwide_gross"].map(floating)

Rename the movie and worldwide gross column

In [47]:
#renaming the movie and worlwide gross column
tn_movie_budgets_df.rename(columns = {'worldwide_gross': 'foreign_gross', 'movie': 'title'}, inplace=True)

In [48]:
tn_movie_budgets_df["release_date"] = tn_movie_budgets_df["release_date"].map(lambda x: int(x.split()[-1]))
tn_movie_budgets_df.rename(columns={"release_date": "year"}, inplace=True)

In [49]:
# changing the production budget to float
tn_movie_budgets_df["production_budget"] = tn_movie_budgets_df["production_budget"].map(floating)

In [50]:
tn_movie_budgets_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5415 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5415 non-null   int64  
 1   year               5415 non-null   int64  
 2   title              5415 non-null   object 
 3   production_budget  5415 non-null   float64
 4   domestic_gross     5415 non-null   float64
 5   foreign_gross      5415 non-null   float64
dtypes: float64(3), int64(2), object(1)
memory usage: 296.1+ KB


#### The movie Dataset

Dealing with nan values

In [51]:
tmdb_df[tmdb_df.duplicated(keep=False)]

genre_ids      id original_language             original_title  \
3      [16, 35, 10751]     862                en                  Toy Story   
10     [16, 35, 10751]     863                en                Toy Story 2   
43         [35, 10749]     239                en           Some Like It Hot   
54       [12, 28, 878]   20526                en               TRON: Legacy   
56     [35, 16, 10751]    9994                en  The Great Mouse Detective   
...                ...     ...               ...                        ...   
26481         [35, 18]  270805                en              Summer League   
26485         [27, 53]  453259                en     Devils in the Darkness   
26504     [27, 35, 27]  534282                en                       Head   
26510             [99]  495045                en                 Fail State   
26511             [99]  492837                en          Making Filmmakers   

       popularity release_date                      title  vote_average  \
3          28.005   1995-11-22                  Toy Story           7.9   
10         22.698   1999-11-24                Toy Story 2           7.5   
43         14.200   1959-03-18           Some Like It Hot           8.2   
54         13.459   2010-12-10               TRON: Legacy           6.3   
56         13.348   1986-07-02  The Great Mouse Detective           7.1   
...           ...          ...                        ...           ...   
26481       0.600   2013-03-18              Summer League           4.0   
26485       0.600   2013-05-15     Devils in the Darkness           3.5   
26504       0.600   2015-03-28                       Head           1.0   
26510       0.600   2018-10-19                 Fail State           0.0   
26511       0.600   2018-04-07          Making Filmmakers           0.0   

       vote_count  
3           10174  
10           7553  
43           1562  
54           4387  
56            769  
...           ...  
26481           3  
26485           1  
26504           1  
26510           1  
26511           1  

[2016 rows x 9 columns]

In [52]:
# investigating more on the duplicates
tmdb_df[tmdb_df["title"] == 'Toy Story']

genre_ids   id original_language original_title  popularity  \
3     [16, 35, 10751]  862                en      Toy Story      28.005   
2473  [16, 35, 10751]  862                en      Toy Story      28.005   

     release_date      title  vote_average  vote_count  
3      1995-11-22  Toy Story           7.9       10174  
2473   1995-11-22  Toy Story           7.9       10174

In [53]:
# dropping the duplicates
tmdb_df.drop_duplicates(inplace=True)

In [54]:
#checking for any other duplicate
tmdb_df.duplicated().any()

False

Dropping columns.

In [55]:
# drop the genre_ids.A significant number of rows have an empty list. 
tmdb_df.drop("genre_ids", axis=1, inplace=True)

In [56]:
# investigating the original_title and the title
tmdb_df[["original_title", "title"]]

# They have similar values in both. We drop original_title
tmdb_df.drop("original_title", axis=1, inplace=True)

In [57]:
#drop the vote_average ,vote_count and popularity
# tmdb_df.drop(["vote_average", "vote_count"], axis=1, inplace=True)

Change the release date to year

In [58]:
tmdb_df.rename(columns={"release_date": "year"}, inplace=True)
tmdb_df["year"] = tmdb_df["year"].map(lambda x: x.split('-')[0])

In [59]:
tmdb_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25497 entries, 0 to 26516
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 25497 non-null  int64  
 1   original_language  25497 non-null  object 
 2   popularity         25497 non-null  float64
 3   year               25497 non-null  object 
 4   title              25497 non-null  object 
 5   vote_average       25497 non-null  float64
 6   vote_count         25497 non-null  int64  
dtypes: float64(2), int64(2), object(3)
memory usage: 1.6+ MB


#### Feature Engineering

Add total revenue column as the sum of domestic gross and foreign gross  to the BOM and The Numbers Dataset

In [60]:
box_office_mojo_df["total_revenue"] = box_office_mojo_df["domestic_gross"] + box_office_mojo_df["foreign_gross"]
box_office_mojo_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

   foreign_gross  year  total_revenue  
0    652000000.0  2010   1.067000e+09  
1    691300000.0  2010   1.025500e+09  
2    664300000.0  2010   9.603000e+08  
3    535700000.0  2010   8.283000e+08  
4    513900000.0  2010   7.526000e+08

In [61]:
tn_movie_budgets_df["total_revenue"] = tn_movie_budgets_df["domestic_gross"] + tn_movie_budgets_df["foreign_gross"]
tn_movie_budgets_df.head()

id  year                                        title  production_budget  \
0   1  2009                                       Avatar        425000000.0   
1   2  2011  Pirates of the Caribbean: On Stranger Tides        410600000.0   
2   3  2019                                 Dark Phoenix        350000000.0   
3   4  2015                      Avengers: Age of Ultron        330600000.0   
4   5  2017            Star Wars Ep. VIII: The Last Jedi        317000000.0   

   domestic_gross  foreign_gross  total_revenue  
0     760507625.0   2.776345e+09   3.536853e+09  
1     241063875.0   1.045664e+09   1.286728e+09  
2      42762350.0   1.497624e+08   1.925247e+08  
3     459005868.0   1.403014e+09   1.862020e+09  
4     620181382.0   1.316722e+09   1.936903e+09

## Data Analysis

#### Univariate Analysis

The total revenue mean

In [62]:
total_revenue_mean = tn_movie_budgets_df["total_revenue"].mean()
total_revenue_mean

142399275.144229

In [63]:
f"The average total revenue from making movies is {total_revenue_mean}"

'The average total revenue from making movies is 142399275.144229'

Production budget mean

In [64]:
production_budget_mean = tn_movie_budgets_df["production_budget"].mean()

In [65]:
f"The average production budget in  making movies is {production_budget_mean}"

'The average production budget in  making movies is 33308006.085318558'

The studios involved in film making

In [66]:
box_office_mojo_df["studio"].describe()

count     2007
unique     172
top       Uni.
freq       144
Name: studio, dtype: object